### PR TITLE
Make the queue size configurable

### PR DESCRIPTION
**Pull request by alde**

### CODE REVIEW

File: scheduler/scheduler_test.go [hunk 1]
```diff
@@ -71,6 +72,14 @@ func TestScheduler(t *testing.T) {
 		Convey("createEremeticScheduler", func() {
 			s := createEremeticScheduler()
 			So(s.tasksCreated, ShouldEqual, 0)
+			So(cap(s.tasks), ShouldEqual, 100)
+		})
+
+		Convey("createEremeticScheduler with configured channel size", func() {
+			MaxQueueSize = 200
```
Comment (keis) [inline scheduler/scheduler_test.go]: When making this runtime configurable I don't think it makes sense to keep this as a global variable. lets add a paramater to `createEremeticScheduler`?

I've seen this pattern, using a options struct, a few times and I quite like it
```
createEremeticScheduler(SchedulerOptions{
    QueueSize: 200
})
```

Comment (alde) [inline scheduler/scheduler_test.go]: Hm, that's a nice pattern, I like it.

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -19,7 +20,13 @@ import (
 var (
 	defaultFilter = &mesos.Filters{RefuseSeconds: proto.Float64(10)}
 	maxRetries    = 5
-	ErrQueueFull  = errors.New("task queue is full")
+
+	// ErrQueueFull is returned in the event of a full queue. This allows the caller
+	// to handle this as he sees fit.
```
Comment (keis) [inline scheduler/scheduler.go]: /s/he/they

:see_no_evil:

**Comment by keis** [comment]
[LGTM](http://lgtm.in/i/2ZYYZ4rWp)

**Pull request merged**